src/scry/analyzer.cr

```diff
@@ -5,12 +5,22 @@ require "./publish_diagnostic"
 
 module Scry
   struct Analyzer
+    @@source = [] of String
```

Comment:
We can leave this for now, but this seems like a likely place for a race condition. It may be time to consider re-architecting the Analyzer.